feathr_project/feathr/utils/job_utils.py

```diff
@@ -1,77 +1,187 @@
+        )
+
+    if client.spark_runtime == "local":
+        if local_cache_path is not None:
+            logger.warning(
+                "In local spark mode, the result files are expected to be stored at a local storage and thus `local_cache_path` argument will be ignored."
+            )
+        local_cache_path = res_url
+    elif client.spark_runtime == "databricks":
+        if res_url.startswith("dbfs:"):
+            logger.warning(
+                "Result files are already in DBFS and thus `local_cache_path` will be ignored."
+            )
+            local_cache_path = res_url
+        else:
+            # if local_cache_path params is not provided then create a temporary folder
+            if local_cache_path is None:
+                # We'll just use the name of a local TemporaryDirectory to cache the data into DBFS.
+                local_cache_path = TemporaryDirectory().name
+
+            # Databricks uses "dbfs:/" prefix for spark paths
+            if not local_cache_path.startswith("dbfs:"):
+                local_cache_path = str(Path("dbfs:", local_cache_path.lstrip("/")))
+    # TODO elif azure_synapse
+
+    if local_cache_path != res_url:
+        logger.info(f"{res_url} files will be downloaded into {local_cache_path}")
+        client.feathr_spark_launcher.download_result(result_path=res_url, local_folder=local_cache_path)
 
-    # use user provided format, if there isn't one, then otherwise use the one provided by the job; 
+    # use user provided format, if there isn't one, then otherwise use the one provided by the job;
     # if none of them is available, "avro" is the default format.
-    format: str = format or client.get_job_tags().get(OUTPUT_FORMAT, "")
-    if format is None or format == "":
-        format = "avro"
+    data_format: str = data_format or client.get_job_tags().get(OUTPUT_FORMAT, "")
+    if data_format is None or data_format == "":
+        data_format = "avro"
 
-    # if local_folder params is not provided then create a temporary folder
-    if local_folder is not None:
-        local_dir_path = local_folder
+    result_df = None
+
+    if spark is not None:
+        result_df = spark.read.format(data_format).load(local_cache_path)
     else:
-        tmp_dir = tempfile.TemporaryDirectory()
-        local_dir_path = tmp_dir.name
-        
-    client.feathr_spark_launcher.download_result(result_path=res_url, local_folder=local_dir_path)
-    dataframe_list = []
-    # by default the result are in avro format
-    if format.casefold()=="parquet":
-        files =  glob.glob(os.path.join(local_dir_path, '*.parquet'))
+        result_df = _read_files_to_pandas_df(
+            dir_path=local_cache_path.replace("dbfs:", "/dbfs"),  # replace to python path if spark path is provided.
+            data_format=data_format,
+        )
+
+    return result_df
+
+
+def _read_files_to_pandas_df(dir_path: str, data_format: str = "avro") -> pd.DataFrame:
```

Comment:
Add some constrains to validate "dir_path" will make this function more robust. Given the above 
```
dir_path=local_cache_path.replace("dbfs:", "/dbfs"),  # replace to python path if spark path is provided.
```

Reply:
If you're okay, I would leave this function as it is (without extra check of the path) without having an extra check. As the name's prefix suggested (starting with an underscore `_`), this function is intended to be hidden and only used by public functions of this module. The public function who calls this function already checks if the path contains `dbfs`.